sympy/calculus/tests/test_util.py

```diff
@@ -534,5 +534,48 @@ def test_issue_16469():
     f = abs(x)
     assert function_range(f, x, S.Reals) == Interval(0, oo, False, True)
 
+
+def test_argmax_real():
+    from sympy.sets.fancysets import ImageSet
+    from sympy.core.function import Lambda
+
+    x = Symbol('x')
+    y = Symbol('y', real = True)
+
+    raises (ValueError, lambda : argmax_real(y, x, S.Reals))
+    raises (ValueError, lambda: argmax_real(x*y, domain = S.Reals))
+    assert argmax_real(x - 2, domain=Interval(2, 3)) == FiniteSet(3)
+    assert argmax_real(x**2 - 2*x + 1, domain=Interval(-oo, -3)) == S.EmptySet
+    assert argmax_real(x**2, domain=Interval(-oo, 4)) == S.EmptySet
+    assert argmax_real(-(log(x)), x, Interval(0.02, 10)) == FiniteSet(0.02)
+    assert argmax_real(sin(x), x, S.Reals) == \
+        ImageSet(Lambda(x, 2*x*pi + pi/2), S.Integers)
+    assert argmax_real(x**2, x, Interval(-3, 3)) == FiniteSet(-3, 3)
+    assert argmax_real(exp(y), y, Interval(-oo, 10)) == FiniteSet(10)
+    assert argmax_real(exp(y), y, S.Reals) == S.EmptySet
+
+
+def test_argmin_real():
+    from sympy.sets.fancysets import ImageSet
+    from sympy.core.function import Lambda
+
+    x = Symbol('x')
+    y = Symbol('y', real = True)
+
+    raises (ValueError, lambda: argmin_real(y, x, S.Reals))
+    assert argmin_real(x**2 - 2*x + 1, domain = Interval(-oo, -3)) == FiniteSet(-3)
+    assert argmin_real(x**2 - 2*x + 1, domain = S.Reals) == FiniteSet(1)
+    assert argmax_real(sin(x), x, S.Reals) == \
+        ImageSet(Lambda(x, 2*x*pi + pi/2), S.Integers)
+    assert argmin_real(log(x), x, Interval(1, 10)) == FiniteSet(1)
+    assert argmin_real(exp(y), y, Interval(-10, 10)) == FiniteSet(-10)
+    assert argmin_real(-x**2 + 4, x, Interval(-2, 2)) == FiniteSet(-2, 2)
+    assert argmin_real(exp(y)) == S.EmptySet
+    assert argmin_real(abs(x), x, S.Reals) == FiniteSet(0)
+    assert argmin_real(abs(2*x-1), domain=S.Reals) == FiniteSet(1/2)
```

Comment:
Any test cases for domains like, `{1, 2, 3}` i.e., those which are finite or countably-infinite?